vrecord

```diff
@@ -925,6 +935,15 @@ if [ -z "${duration}" ] ; then
     read duration
 fi
 
+# confirm legal value for diration
+if [ -n "${duration}" ]; then
+real_regexp="[+-]?([0-9]+|[0-9]+\.[0-9]*|\.[0-9]+)"
+if [[ $duration =~ ^$real_regexp$ ]]; then
+is_number=yes
+else echo "Illegal value for recording time. Input must only be numbers." && exit
+fi
+fi
+
```

Comment:
Sorry, my programming style comes from the last millennium: the first computer I built, 41 years ago, had 256 B (no letter missing!) of memory...